src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -677,11 +677,11 @@ namespace QgsWms
 
     // configure each layer with opacity, selection filter, ...
     bool updateMapExtent = mWmsParameters.bbox().isEmpty();
-    Q_FOREACH ( QgsMapLayer *layer, layers )
+    for ( QgsMapLayer *layer : layers )
     {
       checkLayerReadPermissions( layer );
 
-      Q_FOREACH ( QgsWmsParametersLayer param, params )
+      for ( const QgsWmsParametersLayer param : params )
```

Comment:
I think it should be `const QgsWmsParametersLayer &param` ( the `&` is missing).